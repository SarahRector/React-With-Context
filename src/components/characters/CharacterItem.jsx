import React from 'react';
import PropTypes from 'prop-types';
import { useThemePicker } from '../../hooks/themes';
import '../page-theme/Theme.css';

const CharacterItem = ({ name, image }) => {
  const { theme } = useThemePicker();

  return (
    <figure className={theme}>
      <img src={image} alt={name} width="150px" />
      <figcaption>{name}</figcaption>
    </figure>
  );
};

CharacterItem.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default CharacterItem;
