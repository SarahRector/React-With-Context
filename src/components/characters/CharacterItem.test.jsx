import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterItem from './CharacterItem';

describe('CharacterItem component', () => {
  afterEach(() => cleanup());
  it('renders CharacterItem', () => {
    const { asFragment } = render(<CharacterItem
      name="Arnold"
      image="test.png" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
