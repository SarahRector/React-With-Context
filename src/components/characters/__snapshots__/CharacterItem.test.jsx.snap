// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem component renders CharacterItem 1`] = `
<DocumentFragment>
  <figure
    class="light"
  >
    <img
      alt="Arnold"
      src="test.png"
      width="150px"
    />
    <figcaption>
      Arnold
    </figcaption>
  </figure>
</DocumentFragment>
`;
